### PR TITLE
Overhaul CI and add an MSRV

### PR DESCRIPTION
**Pull request by DJMcNab**
Adds an MSRV, and adapts the CI from https://github.com/linebender/glazier/pull/149

Also prepares for merge queues

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,24 +1,156 @@
...
+
+      - name: install stable toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_STABLE_VER }}
+          targets: wasm32-unknown-unknown
+          components: clippy
+
+      - name: cargo clippy (wasm)
+        run: cargo clippy --all-targets --target wasm32-unknown-unknown -- -D warnings
```
Comment (Philipp-M) [inline .github/workflows/ci.yml:137]: Is `--all-targets` necessary here (when it is already done in other jobs?)?

**Review by Philipp-M — APPROVED** [review]
Looks good to me, thanks

File: .github/workflows/ci.yml (hunk 1 above)
Comment (DJMcNab) [reply .github/workflows/ci.yml:137]: I think so.

This ensures that any tests, benchmarks and examples check for wasm (even though there currently are none of those)

This is cargo having two different meanings of the word "target" (or well at least three, if you include the target directory).

**Review by dfrg — APPROVED** [review]
LGTM, thanks for cleaning this up!

**Pull request merged**